### DISCUSSION TITLE
Feature Request: Load clipboard data into duckdb

### DISCUSSION THREAD

**Discussion by Stitch10925** [duckdb/duckdb#14939] [Ideas]
It would be nice if there was a way to load data from the clipboard into duckdb.

Background is the following:

I often have to extract data from JSON data from API's. Now I have to call the API, copy the data into a file, then load the file into duckdb for easy sorting and extraction.

It would be nice if I could skip the file part and just open duckdb and load the data into memory from the clipboard. It could be via a method which requires you to pass the type of data (JSON in this case) so that it knows how to analyze it.

This could then possibly be expanded to copied CSV or Excel cells, etc. 

Seems like a powerful feature.

**Comment by jgrant-the-giant** [comment]
One option is to transfer through pandas instead, if using python. 
Passing custom args can be helpful for identifying date columns.

```
from duckdb import DuckDBPyConnection
import pandas as pd

def clipboard_to_ddb_table(connection:DuckDBPyConnection, table_name:str, **args):
    """Copies clipboard into duckdb connection.main.table_name

    Args:
        connection (DuckDBPyConnection)
        table_name (str)
    """

    _tempdata = pd.read_clipboard(**args)
    connection.sql(f"create or replace table {table_name} as select * from _tempdata")
```

**Comment by FrancoisLepoutre** [comment]
As @jgrant-the-giant mentioned, you have solutions once you move your data through pandas. 

Other "_fat python client_" solutions, though not as fat as this biggie, pandas, include moving data through pyarrow. I tested and found the later quite suitable. Googling for duckdb python BytesIO StringIO read_csv may lead you to other alternatives including using fspec based-ones. 

I certainly understand you may wish a straight clipboard import from duckdb. 

- On the one hand, doesn't that looks like a risky solution in terms of architecture, memory handling and possibly security pov? I'd better leave clipboard out of the db engine loop.
- On the other hand, feeding our great duck from smallish csv-s, spreadsheets or pieces of json straight from clipboard may make sense as an additional comfort! 

I'd be glad to read feedback from other duckdb users in this respect on "_What do they use for such tasks à la MS-Excel such as clipboarding data into duckdb?_" if any.  These are certainly marginal use cases since most duckdb tasks are certainly not the ones financiers would run on a spreadsheet! But you happen to need importing limited datasets at times anyway.
